patches/tModLoader/Terraria/ModLoader/ModWaterStyle.cs

```diff
@@ -60,6 +60,23 @@ public abstract class ModWaterStyle : ModTexturedType
 		public virtual Color BiomeHairColor() {
 			return new Color(28, 216, 94);
 		}
+
+		/// <summary>
+		/// Returns the texture to be used when drawing rain of this water type.
+		/// <br/>Default uses the vanilla rain texture.
+		/// </summary>
+		public virtual Texture2D GetRainTexture() {
+			return TextureAssets.Rain.Value;
+		}
```

Comment:
This should return Asset<Texture2D> (and the rendering code might need to halt if the .Value of the asset is null.)

Reply:
That makes sense. I'll make it skip the raindrop if the asset value is null. Presumably we still want to render vanilla rain.

Reply:
Should it though? What if you programmatically generate a Texture2D? Cheatsheet does this for some of its stuff afaik

Reply:
It's a quick revert to go back to Texture2D, it being more dynamic is an argument in favor. In defense of using Asset, the rest of the textures referenced in this file are all assets in some static array.

Reply:
I think this is good, dynamic assets are a more generally required todo.